include/swift/AST/LocalizationFormat.h

```diff
@@ -38,6 +38,10 @@ namespace swift {
 enum class DiagID : uint32_t;
 
 namespace diag {
+/// A vector that keeps track of the diagnostics IDs that are available in YAML
+/// and not available in `.def` files.
+extern std::vector<std::string> removedDiagnosticsIDs;
```

Comment:
I think this belongs only to `YAMLLocalizationProducer` let's avoid creating global arrays.

Reply:
I think the same, but since I'm pushing to the vector in a non-member function https://github.com/apple/swift/pull/33337/files#diff-594606b3ff2e7321e3fac5cc36c08dcbR191
as far as I know, I can't call a class field i.e. `removedDiagnosticsIDs` from there. Please correct me if I'm mistaken.

Reply:
What I mean is - we need to find a way how to make it work so it's encapsulated even if it means changes the way it correctly works :)

Reply:
I figured it out! Going to test and push the changes in a moment. :)